### DISCUSSION TITLE
How to create a list of relations with `fields` list of ids

### DISCUSSION THREAD

**Discussion by fadhilsaheer** [prisma/prisma#9922] [Q&A] [answered]
Hey 👋, I'm working on a project which is a room based chat system, I need to store all users in a room so I'm trying to use this schema.

```prisma
model User {
  id       Int    @id @unique @default(autoincrement())
  email    String @unique
  name     String
  profile  String
  password String
  Room     Room[]
}

model Room {
  id          Int     @id @unique @default(autoincrement())
  roomId      String  @unique
  title       String
  description String
  isPrivate   Boolean
  creator     User    @relation(fields: [creatorId], references: [id])
  creatorId   Int
  users       User[] // i need to get all users in query from here
  usersId     String[] // i need to store user's id here
}
```
I want to get all users in `users` in `Room` model and i only need to store user id in `usersId`

This is the way that I'm trying to store, any other way would also work 😃

**Comment by TasinIshmam** [comment, marked as answer]
Hey @fadhilsaheer  :wave:  

I think there is a much better way to model the relationship between `User` and `Room` for your use case.

Here is what I would do: 

1. Make a `one-to-many` relationship between `User` and `Room` to model rooms created by a User (One user can create many rooms, but a single room only has one creator). Let's call this relation `createdRooms`.
2. Make a `many-to-many` relatioship between `User` and `Room` to model rooms a user has joined (One user can join many rooms and one room can have many users in it). Let's call this relation `joinedRooms`. 

Here is what the updated schema looks like

```prisma
model User {
  id           Int    @id @unique @default(autoincrement())
  email        String @unique
  name         String
  profile      String
  password     String
  createdRooms Room[] @relation("createdRooms")
  joinedRooms  Room[] @relation("joinedRooms")
}

model Room {
  id          Int     @id @unique @default(autoincrement())
  roomId      String  @unique
  title       String
  description String
  isPrivate   Boolean
  creator     User    @relation("createdRooms", fields: [creatorId], references: [id])
  creatorId   Int
  joinedUsers       User[]  @relation("joinedRooms")
}
```

When modeling the relation this way, you don't need to store `UserId` fields explicitly in the `Room` model. Since you have created a `many-to-many` relation between `Room` and `User`, Prisma will automatically create a relation table (also known as a join table) between the two fields which will map all `user` records that are in a room (and vice-versa). You can learn more about this in the [many-to-many concept guide](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#relation-tables) in the docs. 

Btw, if I have made a mistake in understanding your problem, or if there is a specific query that you need help with, please feel free to clarify!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
